### PR TITLE
Remove StatefulSetStartOrdinal feature gate to target stable in 1.31

### PR DESCRIPTION
**Pull request by pwschuurman**
Remove StatefulSetStartOrdinal feature gate to target stable in 1.31

* KEP-3335 Enhancement: https://github.com/kubernetes/enhancements/pull/3335

### CODE REVIEW

**Comment by tengqm** [comment]
Please check your commit for auto-generated contents. Those pages are not supposed to be manually updated.

File: content/en/docs/reference/command-line-tools-reference/kube-apiserver.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/kube-apiserver.md:1]: Please exclude changes to the auto-generated [kube-apiserver reference page](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/#auto-generated-edit-notice). During the release process, this file's content is autogenerated (and then the generated docs are committed into this repo), so you don't need to amend them in this PR.

File: content/en/docs/reference/command-line-tools-reference/kube-controller-manager.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/kube-controller-manager.md:1]: Please exclude changes to the auto-generated [kube-controller-manager reference page](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/#auto-generated-edit-notice). During the release process, this file's content is autogenerated (and then the generated docs are committed into this repo), so you don't need to amend them in this PR.

File: content/en/docs/reference/command-line-tools-reference/kube-proxy.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/kube-proxy.md:1]: Please exclude changes to the auto-generated [kube-proxy reference page](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/#auto-generated-edit-notice). During the release process, this file's content is autogenerated (and then the generated docs are committed into this repo), so you don't need to amend them in this PR.

File: content/en/docs/reference/command-line-tools-reference/kube-scheduler.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/kube-scheduler.md:1]: Please exclude changes to the auto-generated [kube-scheduler reference page](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/#auto-generated-edit-notice). During the release process, this file's content is autogenerated (and then the generated docs are committed into this repo), so you don't need to amend them in this PR.

File: content/en/docs/reference/command-line-tools-reference/kubelet.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/kubelet.md:1]: This file content is generated as part of the release process, and the generated docs are committed into this repository separately. Hence, there's no need to modify it within this pull request.

File: content/en/docs/reference/kubernetes-api/workload-resources/stateful-set-v1.md
Comment (dipesh-rawat) [inline content/en/docs/reference/kubernetes-api/workload-resources/stateful-set-v1.md:1]: Please exclude changes to the auto-generated [StatefulSet reference page](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/stateful-set-v1//#auto-generated-edit-notice). During the release process, this file content is autogenerated (and then the generated docs are committed into this repo), so you don't need to amend them in this PR.

**Review by dipesh-rawat — COMMENTED** [review]
@pwschuurman, We should consider setting the `dev-1.31` branch as the target branch rather than the `main` branch for this pull request. Since the change is related to the feature targeting `stable` in 1.31.

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 1]
```diff
@@ -178,13 +178,11 @@ will also add a pod label with this index: `apps.kubernetes.io/pod-index`.
 
 ### Start ordinal
 
-{{< feature-state for_k8s_version="v1.27" state="beta" >}}
+{{< feature-state for_k8s_version="v1.31" state="stable" >}}
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:181]: I suggest:
```suggestion
{{< feature-state feature_gate_name="StatefulSetStartOrdinal" >}}
```

which auto updates, and is our current good practice.

**Comment by pwschuurman** [comment]
> @pwschuurman, We should consider setting the `dev-1.31` branch as the target branch rather than the `main` branch for this pull request. Since the change is related to the feature targeting `stable` in 1.31.

Thanks, updated to `dev-1.31` branch.

**Comment by tengqm** [comment]
/hold
The upstream change is not merged yet.

**Comment by pwschuurman** [comment]
Hi @tengqm, the upstream change https://github.com/kubernetes/kubernetes/pull/125374 has been merged.

**Review by mengjiao-liu — COMMENTED** [review]
/lgtm

**Pull request merged**